src/vm/database/structures.rs

```diff
@@ -407,6 +407,29 @@ impl<'db, 'conn> STXBalanceSnapshot<'db, 'conn> {
         };
     }
 
+    pub fn extend_lock_v2(&mut self, unlock_burn_height: u64) {
+        let unlocked = self.unlock_available_tokens_if_any();
+        if unlocked > 0 {
+            debug!("Consolidated after extend-token-lock");
+        }
+
+        if !self.has_locked_tokens() {
+            // caller needs to have checked this
+            panic!("FATAL: account does not have locked tokens");
+        }
+
+        if unlock_burn_height <= self.burn_block_height {
+            // caller needs to have checked this
+            panic!("FATAL: cannot set a lock with expired unlock burn height");
+        }
+
+        self.balance = STXBalance::LockedPoxTwo {
```

Comment:
i'm curious why it is right to overwrite this.. is there a test that calls this twice and three times?

Reply:
Added a comment to this function -- this method is invoked by PoX2 as a result of `stack-extend` or `delegate-stack-extend`. The account structure reflects whether or not the account was locked by PoX1 or PoX2 -- this method will upgrade an account to PoX2. The two `pox_2_tests` attempt to invoke PoX functions in PoX1 after this has been called -- the expectation is that if PoX1 attempt to re-lock an account that was locked by PoX2 already, then it will abort with a runtime exception (which is checked in the tests).

Reply:
Added in bbd826d4bde81aee6c26f134439b9d73d8885e34